### PR TITLE
EWPP-3296: Theme the Media paragraph.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

File: oe_theme.theme [hunk 1]
```diff
@@ -2204,3 +2204,91 @@ function oe_theme_preprocess_oe_theme_helper_in_page_navigation__node__oe_person
...
+function oe_theme_preprocess_paragraph__oe_av_media(array &$variables): void {
+  /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+  $paragraph = $variables['paragraph'];
+
+  // Bail out if there is no media present.
+  if ($paragraph->get('field_oe_media')->isEmpty()) {
+    return;
+  }
+
+  $cacheability = CacheableMetadata::createFromRenderArray($variables);
```
Comment (nagyad) [inline oe_theme.theme:2220]: Why to create this here if you bail out after when the media entity is not available anymore in line 2226?

File: oe_theme.theme [hunk 2]
```diff
@@ -2204,3 +2204,91 @@ function oe_theme_preprocess_oe_theme_helper_in_page_navigation__node__oe_person
...
+
+  // If it's an image media, render it and assign it to the image variable.
+  if ($source instanceof MediaAvPortalPhotoSource || $source instanceof Image) {
+    $thumbnail = $media->get('thumbnail')->first();
+    $variables['image'] = ImageValueObject::fromStyledImageItem($thumbnail, 'oe_theme_medium_no_crop');
+    $cacheability->applyTo($variables);
+    return;
+  }
+
+  $cacheability->applyTo($variables);
```
Comment (nagyad) [inline oe_theme.theme]: You already apply the cache of the $variables in line 2275 and after that you don't add anything to it, so this is not needed.

File: oe_theme.theme [hunk 3]
```diff
@@ -2204,3 +2204,91 @@ function oe_theme_preprocess_oe_theme_helper_in_page_navigation__node__oe_person
...
+      $variables['ratio'] = str_replace('_', '-', $ratio);
+    }
+    return;
+  }
+
+  // If it's an image media, render it and assign it to the image variable.
+  if ($source instanceof MediaAvPortalPhotoSource || $source instanceof Image) {
+    $thumbnail = $media->get('thumbnail')->first();
+    $variables['image'] = ImageValueObject::fromStyledImageItem($thumbnail, 'oe_theme_medium_no_crop');
+    $cacheability->applyTo($variables);
```
Comment (nagyad) [inline oe_theme.theme]: You already apply the cache of the $variables in line 2275 and after that you don't add anything to it, so this is not needed.

**Review by nagyad — CHANGES_REQUESTED** [review]

File: oe_theme.theme (hunk 1 above)
Comment (22Alexandra) [reply oe_theme.theme:2220]: It's created after making sure a media item exists. Why would it be created before?

File: oe_theme.theme (hunk 3 above)
Comment (22Alexandra) [reply oe_theme.theme]: Not really, that's for video media items. After we finish preprocessing the video, we return on 2282.

File: oe_theme.theme (hunk 2 above)
Comment (22Alexandra) [reply oe_theme.theme]: Same as below, on 2275 we apply the cache for the video item and then we return. 
Here we apply the cache that contains only the translated media and it's access which means we don't have a video or an image anymore.

**Review by nagyad — DISMISSED** [review]

File: oe_theme.theme [hunk 4]
```diff
@@ -2204,3 +2204,88 @@ function oe_theme_preprocess_oe_theme_helper_in_page_navigation__node__oe_person
...
+    return;
+  }
+
+  // Get the media source.
+  $source = $media->getSource();
+
+  if ($source instanceof MediaAvPortalVideoSource || $source instanceof OEmbed || $source instanceof Iframe) {
+    if ($source instanceof MediaAvPortalVideoSource) {
+      // Default video aspect ratio is set to 16:9 for AV Portal Video.
+      $variables['ratio'] = '16:9';
```
Comment (imanoleguskiza) [inline oe_theme.theme]: This seems to be messing up with the height of avportal v ideos

**Review by imanoleguskiza — CHANGES_REQUESTED** [review]

**Review by imanoleguskiza — DISMISSED** [review]

**Review by nagyad — DISMISSED** [review]

File: composer.json [hunk 5]
```diff
@@ -50,7 +50,7 @@
         "openeuropa/oe_corporate_countries": "^2.0.0-alpha8",
         "openeuropa/oe_media": "^1.23.1",
         "openeuropa/oe_multilingual": "^1.13",
-        "openeuropa/oe_paragraphs": "^1.19",
+        "openeuropa/oe_paragraphs": "dev-EWPP-3295",
```
Comment (nagyad) [inline composer.json]: Needs merge first.

**Review by nagyad — CHANGES_REQUESTED** [review]

**Review by nagyad — APPROVED** [review]

**Review by hernani — APPROVED** [review]

**Pull request merged**